tmail-backend/mailets/src/main/scala/com/linagora/tmail/mailets/IndexContacts.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.mailet.MailetException;
+import org.apache.mailet.base.GenericMailet;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.guava.GuavaModule;
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.linagora.tmail.james.jmap.EmailAddressContactInjectKeys;
+import com.linagora.tmail.james.jmap.contact.ContactFields;
+import com.linagora.tmail.james.jmap.contact.TmailContactUserAddedEvent;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import spark.utils.StringUtils;
+
+/**
+ * <p><b>IndexContacts</b> allows indexing user contacts locally by dispatching a <code>ContactUserAddedEvent</code> to the <code>eventBus</code>,
+ * in order to index contacts asynchronously.</p>
+ *
+ * <p>This mailet takes as input an attribute containing the user and the contacts to be added to that user.
+ * The input should be in JSON format, and compliant with the format of {@link ContactExtractor.ExtractedContacts}.</p>
+ *
+ * <p>Here is the JSON format:</p>
+ * <pre><code>
+ * {
+ *   "userEmail" : "sender@james.org",
+ *   "emails" : [ "to@james.org", "cc@james.org" ]
+ * }
+ * </code></pre>
+ *
+ * <p>Sample configuration:</p>
+ * <pre><code>
+ * &lt;mailet match="All" class="IndexContacts"&gt;
+ *   &lt;attribute&gt;ExtractedContacts&lt;/attribute&gt;
+ * &lt;/mailet&gt;
+ * </code></pre>
+ */
+public class IndexContacts extends GenericMailet {
+    private static final ImmutableSet<RegistrationKey> NO_REGISTRATION_KEYS = ImmutableSet.of();
+    private static final String EMPTY_STRING = "";
+
+    private final ObjectMapper objectMapper = new ObjectMapper()
+        .registerModule(new GuavaModule());
+
+    private final EventBus eventBus;
+
+    private AttributeName attributeName;
+
+    public IndexContacts(@Named(EmailAddressContactInjectKeys.AUTOCOMPLETE) EventBus eventBus) {
+        this.eventBus = eventBus;
+    }
+
+    @Override
+    public void init() throws MessagingException {
+        attributeName = Optional.ofNullable(getInitParameter("attribute"))
+            .filter(StringUtils::isNotBlank)
+            .map(AttributeName::of)
+            .orElseThrow(
+                () -> new MailetException("No value for `attribute` parameter was provided."));
+    }
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        Mono.justOrEmpty(mail.getAttribute(attributeName))
+            .map(attribute -> attribute.getValue().asAttributeValueOf(String.class))
+            .flatMap(Mono::justOrEmpty)
+            .map(AttributeValue::value)
+            .map(Throwing.function(this::parseUserContacts))
+            .flatMap(Throwing.function(this::dispatchContactsAddedEvent))
+            .onErrorMap(e -> new MailetException("Failed to index contacts", new Exception(e)))
```

Comment:
We are not on a critical feature, so we should WARN on error and avoid propagating the exception